### PR TITLE
Chef provisioning testing

### PR DESCRIPTION
**Pull request by ramereth**
This provides a chef-provisioning environment for testing multi-node interactions between a controller node and a compute node.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,10 @@ bin/*
 
 .kitchen/
 .kitchen.local.yml
+
+# chef-provisioner
```
Comment (jordane) [inline .gitignore]: This should also be added to the `Chefignore` file.

Comment (ramereth) [inline .gitignore]: @jordane done

File: chefignore [hunk 2]
```diff
@@ -92,3 +92,10 @@ Vagrantfile
 # Travis #
 ##########
 .travis.yml
+
+# chef-provisioner
+.chef/*.pem
+.chef/local-mode-cache/
+/clients/
+/nodes/
+/vms/
```
Comment (jordane) [inline chefignore]: We may also want to ignore `controller_compute.rb`, `vagrant_linux.rb`, and the other chef provisioning files.

Which brings up the question, do we want to start putting all of the chef provisioning files in one dir?

Comment (ramereth) [inline chefignore]: We can probably move both of those files into a ``chef-provisioning`` folder. Are there any other files you'd like to move there?

Comment (jordane) [inline chefignore]: `destroy_all.rb`

**Comment by ramereth** [comment]
@jordane I've moved all the related files under ``test/integration/``.

File: Rakefile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+def run_command(command)
+  if File.exist?('Gemfile.lock')
+    sh %(bundle exec #{command})
+  else
+    sh %(chef exec #{command})
+  end
+end
+
+prov_path = 'test/integration/chef-provisioning'
```
Comment (jordane) [inline Rakefile]: Constants are traditionally entirely uppercase.

**Comment by jordane** [comment]
+1

**Pull request merged**